iceoryx_posh/include/iceoryx_posh/internal/roudi/service_registry.hpp

```diff
@@ -58,13 +57,13 @@ class ServiceRegistry
     /// @param[in] serviceDescription, service to be removed
     void remove(const capro::ServiceDescription& serviceDescription) noexcept;
 
-    /// @brief Removes given service description from registry
+    /// @brief Searches for given service description in registry
     /// @param[in] searchResult, reference to the vector which will be filled with the results
     /// @param[in] service, string or wildcard to search for
     /// @param[in] instance, string or wildcard to search for
     void find(ServiceDescriptionVector_t& searchResult,
-              const capro::IdString_t& service = Wildcard,
-              const capro::IdString_t& instance = Wildcard) const noexcept;
+              const cxx::variant<cxx::Wildcard_t, capro::IdString_t>& service,
```

Comment:
Not sure this is possible with `variant` without wrapping it. In any case I prefer overloads in theory, i.e. have a call with *service*, one with *instance*, one with both and one without arguments. This improves dispatching as well and internally the cases are implemented in a switch statement otherwise anyway.

The reasoning is to prefer early dispatching mechanisms.

Reply:
@mossmaurice I decided against because it was not used anywhere, not even tested, and as @MatthiasKillat wrote, it's not that easy with a variant.